### DISCUSSION TITLE
Dragging with right mouse button, middle mouse button, or left and right combined

### DISCUSSION THREAD

**Discussion by sluijs** [pmndrs/use-gesture#336] [Q&A] [answered]
Thanks for the amazing library. I'm using the Vanilla version, and I was wondering if is possible to initiate a `drag` event with the right mouse button or the middle mouse button. My demo code only triggered the event on the left mouse button:

```jsx
class App extends Component {
  constructor(props) {
    super(props)
    this.canvasRef = React.createRef();
  }

  componentDidMount() {
    const { canvasRef: { current: canvas } } = this;
    new DragGesture(canvas, (state) => console.log(state));
  }

  render() {
    return (
      <div className="app">
          <canvas ref={this.canvasRef} />
      </div>
    )
  }
}
```

A workaround might be to use the `MoveGesture` and check for the `buttons` property. (I'm not sure if this is intended behavior, but the `pressed` property is false when the right or middle mouse buttons are pressed).

**Comment by dbismut** [comment, marked as answer]
Hi, indeed the drag gesture won't start if the primary button isn't pressed, and similarly `pressed` is only true when the primary button is pressed. There could be an option where you would specify which button would trigger the drag https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons. 

However I wouldn't recommend using the middle button as a trigger since most trackpads don't have it.

**Reply by sluijs** [reply]
Thanks for your quick response. My app requires the user to have all 3 mouse buttons, which is reasonable for this edge case, but I'll go with the `MoveGesture` then!

**Reply by dbismut** [reply]
Sure! I'd be curious how you manage to make it work with the `buttons` property though. I can see this being a cool addition to the drag gesture options, so let me know when you can ;)
